tidy3d/plugins/design/design.py

```diff
@@ -7,17 +7,24 @@
 
 import pydantic.v1 as pd
 
-from ...components.base import Tidy3dBaseModel, cached_property
+from ...components.base import TYPE_TAG_STR, Tidy3dBaseModel, cached_property
 from ...components.data.sim_data import SimulationData
 from ...components.simulation import Simulation
-from ...web.api.container import BatchData
-from .method import MethodType
-from .parameter import ParameterType
+from ...log import Console, get_logging_console, log
+from ...web.api.container import Batch, BatchData, Job
+from .method import (
+    MethodBayOpt,
+    MethodGenAlg,
+    MethodOptimize,
+    MethodParticleSwarm,
+    MethodType,
+)
+from .parameter import ParameterAny, ParameterInt, ParameterType
 from .result import Result
 
 
 class DesignSpace(Tidy3dBaseModel):
-    """Specification of a design problem / combination of several parameters + algorithm.
+    """Manages all exploration of a parameter space within specified parameters using a supplied search method.
```

Comment:
we're probably going to need a lot more info here since it's the fundamental object for everything. including probably some diagrams and links to notebooks. See Simulation for inspiration

https://github.com/flexcompute/tidy3d/blob/develop/tidy3d/components/simulation.py#L1557-L1658

Reply:
I think:
* A few links to our docs examples using the design plugin.
* A code example, minimum viable example of running the DesignSpace (maybe similar to the first few cells of the Design.ipynb where we sample a 2D gaussian function).
* The design plugin flow diagram